DbtCLI parseRunResults crashes on skipped dbt nodes (State.getDuration: Index -1 out of bounds)

### Describe the issue

With `parseRunResults: true` (default), any **skipped** dbt node makes the worker fail to serialize the result: the task ends FAILED although dbt itself finished, and per-model task runs are lost. Fully green runs are unaffected.

## Error
```
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
    at io.kestra.core.models.flows.State.getDuration(State.java:96)
```
(thrown while serializing `WorkerTaskResult.dynamicTaskRuns`)

## Steps to reproduce
DuckDB, no external warehouse. Reproduces on both Docker and Process runners (backend: Postgres / JDBC).

```yaml
id: dbt_skip_repro
namespace: company.team
tasks:
  - id: dbt
    type: io.kestra.plugin.dbt.cli.DbtCLI
    taskRunner:
      type: io.kestra.plugin.scripts.runner.docker.Docker
    containerImage: ghcr.io/kestra-io/dbt-duckdb:latest
    inputFiles:
      dbt_project.yml: |
        name: demo
        profile: demo
        version: "1.0"
        config-version: 2
        model-paths: ["models"]
      models/m_bad.sql: "select * from this_table_does_not_exist_xyz"
      models/m_child.sql: "{% raw %}select * from {{ ref('m_bad') }}{% endraw %}"
      models/m_ok.sql: "select 1 as id"
    profiles: |
      demo:
        target: dev
        outputs:
          dev: { type: duckdb, path: ":memory:" }
    commands:
      - dbt build
```

`m_bad` errors → `m_child` is skipped → crash.

## Root cause
A skipped node has no `timing` in `run_results.json`, so plugin-dbt builds a terminal `State` with empty `histories` ([ResultParser](https://github.com/kestra-io/plugin-dbt/blob/master/src/main/java/io/kestra/plugin/dbt/ResultParser.java)). Core [`State.getEndDate()`](https://github.com/kestra-io/kestra/blob/master/core/src/main/java/io/kestra/core/models/flows/State.java) then calls `histories.getLast()` with no empty-check.

## Suggested fix
Guard empty `histories` in `State.getEndDate()`, and/or stop emitting empty-history task runs for skipped nodes in plugin-dbt.

## Workaround
`parseRunResults: false`.

### Environment

Kestra v1.3.3 & v1.3.24 · plugin-dbt 1.6.2 · Docker & Process runner · Postgres backend

@loicmathieu 
Sure — sorry for trimming it too much. Full stack trace below, captured on a fresh v1.3.26 standalone (Postgres backend, Docker runner), running the exact repro flow from the issue description. Taken from the execution log download (includes TRACE level).

```
2026-07-08T00:40:39.639201Z ERROR Unable to emit the worker task result to the queue: Unable to serialize the message
2026-07-08T00:40:39.639201Z ERROR Index -1 out of bounds for length 0 (through reference chain: io.kestra.core.runners.WorkerTaskResult["dynamicTaskRuns"]->java.util.ImmutableCollections$ListN[2]->io.kestra.core.models.executions.TaskRun["attempts"]->java.util.ImmutableCollections$List12[0]->io.kestra.core.models.executions.TaskRunAttempt["state"]->io.kestra.core.models.flows.State["duration"])
Index -1 out of bounds for length 0
2026-07-08T00:40:39.639201Z TRACE io.kestra.core.queues.QueueException: Unable to serialize the message
	at io.kestra.jdbc.runner.JdbcQueue.produceFields(JdbcQueue.java:120)
	at io.kestra.runner.postgres.PostgresQueue.produceFields(PostgresQueue.java:37)
	at io.kestra.jdbc.runner.JdbcQueue.produce(JdbcQueue.java:149)
	at io.kestra.jdbc.runner.JdbcQueue.emit(JdbcQueue.java:188)
	at io.kestra.core.queues.QueueInterface.emit(QueueInterface.java:13)
	at io.kestra.worker.DefaultWorker.run(DefaultWorker.java:788)
	at io.kestra.worker.DefaultWorker.handleTask(DefaultWorker.java:378)
	at io.kestra.worker.DefaultWorker.lambda$run$8(DefaultWorker.java:296)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Index -1 out of bounds for length 0 (through reference chain: io.kestra.core.runners.WorkerTaskResult["dynamicTaskRuns"]->java.util.ImmutableCollections$ListN[2]->io.kestra.core.models.executions.TaskRun["attempts"]->java.util.ImmutableCollections$List12[0]->io.kestra.core.models.executions.TaskRunAttempt["state"]->io.kestra.core.models.flows.State["duration"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:400)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:359)
	at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:324)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:765)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContentsUsing(IndexedListSerializer.java:142)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:88)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4938)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsBytes(ObjectMapper.java:4171)
	at io.kestra.jdbc.runner.JdbcQueue.produceFields(JdbcQueue.java:118)
	... 11 more
Caused by: java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Unknown Source)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Unknown Source)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Unknown Source)
	at java.base/java.util.Objects.checkIndex(Unknown Source)
	at java.base/java.util.ArrayList.get(Unknown Source)
	at io.kestra.core.models.flows.State.getEndDate(State.java:123)
	at io.kestra.core.models.flows.State.getDuration(State.java:96)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
	... 33 more
```

--- a/core/src/main/java/io/kestra/core/models/flows/State.java
+++ b/core/src/main/java/io/kestra/core/models/flows/State.java
@@ -119,6 +119,10 @@ public class State {
             return Optional.empty();
         }
 
+        if (this.histories.isEmpty()) {
+            return Optional.empty();
+        }
+
         return Optional.of(this.histories.getLast().getDate());
     }
 

--- a/core/src/test/java/io/kestra/core/models/flows/StateTest.java
+++ b/core/src/test/java/io/kestra/core/models/flows/StateTest.java
@@ -1,0 +1,180 @@
+package io.kestra.core.models.flows;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import io.kestra.core.models.tasks.Task;
+import io.kestra.plugin.core.debug.Return;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class StateTest {
+
+    private static final Instant START = Instant.parse("2024-01-01T00:00:00Z");
+    private static final Instant END = Instant.parse("2024-01-01T01:01:01Z");
+
+    private State terminatedState() {
+        return new State(
+            State.Type.SUCCESS, List.of(
+                new State.History(State.Type.CREATED, START),
+                new State.History(State.Type.SUCCESS, END)
+            )
+        );
+    }
+
+    @Test
+    void shouldReturnEmptyEndDateWhenHistoryIsEmpty() {
+        // Given: a terminated state with no history entries
+        State state = new State(State.Type.SUCCESS, List.of());
+
+        // When / Then: end date is absent rather than throwing
+        assertThat(state.getEndDate()).isEmpty();
+    }
+
+    @Test
+    void shouldGetStartDateFromFirstHistoryEntry() {
+        // Given / When / Then
+        assertThat(terminatedState().getStartDate()).isEqualTo(START);
+    }
+
+    @Test
+    void shouldGetEndDateFromLastHistoryEntryWhenTerminated() {
+        // Given / When / Then
+        assertThat(terminatedState().getEndDate()).contains(END);
+    }
+
+    @Test
+    void shouldGetEmptyEndDateWhenNotTerminatedNorPaused() {
+        // Given
+        State state = new State(State.Type.RUNNING, List.of(new State.History(State.Type.RUNNING, START)));
+
+        // When / Then
+        assertThat(state.getEndDate()).isEmpty();
+    }
+
+    @Test
+    void shouldGetEndDateFromLastHistoryEntryWhenPaused() {
+        // Given
+        State state = new State(
+            State.Type.PAUSED, List.of(
+                new State.History(State.Type.CREATED, START),
+                new State.History(State.Type.PAUSED, END)
+            )
+        );
+
+        // When / Then
+        assertThat(state.getEndDate()).contains(END);
+    }
+
+    @Test
+    void shouldGetDurationBetweenStartAndEndWhenTerminated() {
+        // Given / When / Then
+        assertThat(terminatedState().getDuration()).contains(Duration.between(START, END));
+    }
+
+    @Test
+    void shouldGetEmptyDurationWhenNotTerminated() {
+        // Given
+        State state = new State(State.Type.RUNNING, List.of(new State.History(State.Type.RUNNING, START)));
+
+        // When / Then
+        assertThat(state.getDuration()).isEmpty();
+    }
+
+    @Test
+    void shouldGetPersistedDurationWhenTerminated() {
+        // Given / When / Then
+        assertThat(terminatedState().getDurationOrComputeIt()).isEqualTo(Duration.between(START, END));
+    }
+
+    @Test
+    void shouldComputeDurationOnTheFlyWhenNotTerminated() {
+        // Given
+        Instant start = Instant.now().minusSeconds(5);
+        State state = new State(State.Type.RUNNING, List.of(new State.History(State.Type.RUNNING, start)));
+
+        // When / Then
+        assertThat(state.getDurationOrComputeIt()).isCloseTo(Duration.ofSeconds(5), Duration.ofSeconds(3));
+    }
+
+    @Test
+    void shouldFormatHumanDurationAsHoursMinutesSeconds() {
+        // Given / When / Then
+        assertThat(terminatedState().humanDuration()).isEqualTo("01:01:01.000");
+    }
+
+    @Test
+    void shouldGetMaxDateFromLastHistoryEntry() {
+        // Given / When / Then
+        assertThat(terminatedState().maxDate()).isEqualTo(END);
+    }
+
+    @Test
+    void shouldGetMaxDateAsNowWhenHistoryIsEmpty() {
+        // Given
+        Instant before = Instant.now();
+
+        // When
+        Instant maxDate = new State(State.Type.RUNNING, List.of()).maxDate();
+
+        // Then
+        assertThat(maxDate).isBetween(before, Instant.now());
+    }
+
+    @Test
+    void shouldGetMinDateFromFirstHistoryEntry() {
+        // Given / When / Then
+        assertThat(terminatedState().minDate()).isEqualTo(START);
+    }
+
+    @Test
+    void shouldGetMinDateAsNowWhenHistoryIsEmpty() {
+        // Given
+        Instant before = Instant.now();
+
+        // When
+        Instant minDate = new State(State.Type.RUNNING, List.of()).minDate();
+
+        // Then
+        assertThat(minDate).isBetween(before, Instant.now());
+    }
+
+    @Test
+    void shouldFailWithSuccessWhenAllowFailureAndAllowWarning() {
+        // Given
+        Task task = Return.builder().allowFailure(true).allowWarning(true).build();
+
+        // When / Then
+        assertThat(State.Type.fail(task)).isEqualTo(State.Type.SUCCESS);
+    }
+
+    @Test
+    void shouldFailWithWarningWhenOnlyAllowFailure() {
+        // Given
+        Task task = Return.builder().allowFailure(true).allowWarning(false).build();
+
+        // When / Then
+        assertThat(State.Type.fail(task)).isEqualTo(State.Type.WARNING);
+    }
+
+    @Test
+    void shouldFailWithFailedWhenNotAllowFailure() {
+        // Given
+        Task task = Return.builder().allowFailure(false).build();
+
+        // When / Then
+        assertThat(State.Type.fail(task)).isEqualTo(State.Type.FAILED);
+    }
+
+    @Test
+    void shouldFailWithFailedWhenNotAllowFailureEvenIfAllowWarning() {
+        // Given
+        Task task = Return.builder().allowFailure(false).allowWarning(true).build();
+
+        // When / Then
+        assertThat(State.Type.fail(task)).isEqualTo(State.Type.FAILED);
+    }
+}

